### PR TITLE
sg: when in CI we do not need to prompt for an identity

### PR DESCRIPTION
**Pull request by burmudar**
There are cases  when we use SG in CI and then we do not want to prompt for identity

## Test plan
CI, unit tests and tested locally

### CODE REVIEW

File: dev/sg/internal/analytics/oauth.go [hunk 1]
```diff
@@ -44,6 +46,11 @@ func InitIdentity(ctx context.Context, out *std.Output, sec secretStore) error {
 
 	whoamiPath := path.Join(sgHome, "whoami.json")
 
+	// In CI who do not have to prompt for an identity
```
Comment (burmudar) [inline dev/sg/internal/analytics/oauth.go:49]:
```suggestion
	// In CI we do not have to prompt for an identity
```

File: dev/sg/internal/analytics/analytics_test.go [hunk 2]
```diff
@@ -33,6 +33,16 @@ func TestGetEmail(t *testing.T) {
 		}
 	})
 
+	t.Run("when CI=true identity is ci@sourcegraph.com", func(t *testing.T) {
+		os.Setenv("CI", "true")
+		t.Cleanup(func() {
+			os.Unsetenv("CI")
```
Comment (burmudar) [inline dev/sg/internal/analytics/analytics_test.go:39]: need to be careful with this ...

**Review by jhchabran — COMMENTED** [review]
Why don't we handle that case at the toplevel? For example, we could say that the flag that disables it is always enabled it we see `CI=true` ?

**Comment by burmudar** [comment]
> Why wouldn't we handle that case at the toplevel? I feel it would be simpler For example, we could say that the flag that disables it is always enabled it we see `CI=true` ? 

Yeah I think is a good point tbh. Otherwise it becomes really nasty in the test to enable, disable and restore the CI env var

File: dev/sg/main.go [hunk 3]
```diff
@@ -162,7 +162,11 @@ var sg = &cli.App{
 		if bashCompletionsMode {
 			return nil
 		}
-
+		//
```
Comment (burmudar) [inline dev/sg/main.go:165]:
```suggestion
```

**Review by jhchabran — APPROVED** [review]

**Pull request merged**